### PR TITLE
test: Git integration tests

### PR DESCRIPTION
**Pull request by nidhi-nair**
Fixes #`Issue Number`  
_or_  
Fixes `Issue URL`
> [!WARNING]  
> _If no issue exists, please create an issue first, and check with the maintainers if the issue is valid._

## Automation

/ok-to-test tags=""

### :mag: Cypress test results

> [!WARNING]
> Tests have not run on the HEAD bebfa2a10e292c5664900d43efc838983ddb5616 yet
> <hr>Mon, 23 Dec 2024 08:36:45 UTC

## Communication
Should the DevRel and Marketing teams inform users about this change?
- [ ] Yes
- [ ] No

## Summary by CodeRabbit

- **New Features**
	- Introduced a method to set the name of artifacts.
	- Enhanced error handling for branch deletion and merge operations.
	- Added integration tests for validating Git workflows, including artifact connection, repository state verification, and branch management.
	- Implemented utilities for managing artifact tests and Git server lifecycle.

- **Bug Fixes**
	- Improved error management during branch operations.

- **Tests**
	- Created new integration test suite and utility classes for Git-related functionalities.

### CODE REVIEW

File: app/server/appsmith-server/src/test/utils/com/appsmith/server/git/GitArtifactTestUtils.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+@Component
+public class GitArtifactTestUtils<T extends Artifact> {
+
+    @Autowired
+    LayoutActionService layoutActionService;
+    @Autowired
+    PluginService pluginService;
+
+    Mono<Void> createADiff(Artifact artifact) {
```
Comment (sondermanish) [inline app/server/appsmith-server/src/test/utils/com/appsmith/server/git/GitArtifactTestUtils.java:28]: let's add artifactType as param? can defer the implementation later?

File: app/server/appsmith-server/src/test/utils/com/appsmith/server/git/GitServerInitializerExtension.java [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+                            "read_only", false))
+                        .retrieve()
+                        .toBodilessEntity();
+                })
+            .block();
+
+    }
+
+    @Override
+    public void afterEach(ExtensionContext extensionContext) {
```
Comment (sondermanish) [inline app/server/appsmith-server/src/test/utils/com/appsmith/server/git/GitServerInitializerExtension.java:99]: QQ: this after each is basically after each test class, right?

File: app/server/appsmith-server/src/test/it/com/appsmith/server/git/GitBranchesIT.java [hunk 3]
```diff
@@ -0,0 +1,584 @@
...
+        //  - Is it possible to use controller layer here? Might help with also including web filters in IT
+        Artifact artifact = commonGitService.connectArtifactToGit(artifactId, connectDTO, ORIGIN, gitContext.getArtifactType())
+            .block();
+
+        assertThat(artifact).isNotNull();
+
+        ArtifactType artifactType = artifact.getArtifactType();
+        GitArtifactMetadata artifactMetadata = artifact.getGitArtifactMetadata();
+        GitArtifactHelper<?> artifactHelper = gitArtifactHelperResolver.getArtifactHelper(artifactType);
+        Path repoSuffix = artifactHelper.getRepoSuffixPath(
```
Comment (sondermanish) [inline app/server/appsmith-server/src/test/it/com/appsmith/server/git/GitBranchesIT.java:178]: TODO for me: convert this into a DTO

File: app/server/appsmith-server/src/test/it/com/appsmith/server/git/GitBranchesIT.java [hunk 4]
```diff
@@ -0,0 +1,584 @@
...
+        GitBranchDTO barBranchDTO = new GitBranchDTO();
+        barBranchDTO.setBranchName("bar");
+        Artifact barArtifact = commonGitService.createBranch(fooArtifactId, barBranchDTO, artifactType).block();
+        assertThat(barArtifact).isNotNull();
+
+        String barArtifactId = barArtifact.getId();
+        GitArtifactMetadata barMetadata = barArtifact.getGitArtifactMetadata();
+        assertThat(barMetadata.getBranchName()).isEqualTo("bar");
+
+        try (Git git = Git.open(path.toFile())) {
```
Comment (sondermanish) [inline app/server/appsmith-server/src/test/it/com/appsmith/server/git/GitBranchesIT.java:392]: seems like we can extract a method out of it?

Comment (nidhi-nair) [reply app/server/appsmith-server/src/test/it/com/appsmith/server/git/GitBranchesIT.java:392]: Yes I'm hoping to find more patterns as we write ITs and then work on optimizing this code.

File: app/server/appsmith-server/src/test/utils/com/appsmith/server/git/GitServerInitializerExtension.java (hunk 2 above)
Comment (nidhi-nair) [reply app/server/appsmith-server/src/test/utils/com/appsmith/server/git/GitServerInitializerExtension.java:99]: That's right

**Review by sondermanish — APPROVED** [review]

**Pull request merged**